stackstorm-nexus3/actions/lib/base.py

```diff
@@ -0,0 +1,178 @@
+import sys
+from exception import *
+from nexuscli.repository import Repository
+from nexuscli.nexus_client import NexusClient
+from nexuscli.exception import *
+
+from st2common.runners.base_action import Action
+
+__all__ = [
+    'BaseAction'
+]
+
+REQUIRED_FIELDS = ['url', 'user', 'password']
+
+
+class BaseAction(Action):
+
+    def __init__(self, config):
+        super(BaseAction, self).__init__(config)
+        self.nexus_profiles = self.config.get('profiles', {})
+        self.nexus_client = None
+        self.dial_config = None
+
+    def validate_config(self, c_profile):
+        """ Validate connection configuration
+
+        """
+        validation_errors = []
+        is_valid = True
+        for field in REQUIRED_FIELDS:
+            if c_profile.get(field, None) == None:
+                validation_errors.push("%s if missing" % field)
+                is_valid = False
+
+        return (is_valid, validation_errors)
+
+    def get_connection_defaults(self):
+        """ default properties
+
+        """
+        return {
+            "url": self.config.get('url', None),
+            "verify": self.config.get('verify', True),
+            "user": self.config.get('user', None),
+            "password": self.config.get('password', None)
+        }
+
+    def init_config(self, profile=None):
+        """ Get nexus3 server connection string
+
+        """
+        self.dial_config = self.get_connection_defaults()
+        if profile is not None:
+            profile = profile.strip()
+            if len(profile) == 0:
+                profile = None
+
+        # use 'default_profile' if 'config_profile' is empty
+        if profile is None:
+            profile = self.config.get('default_profile', None)
+
+        # Use defaults options, if neither config_profile nor default_profile is provided
+        if profile is None:
+            self.logger.info("using defaults connection values")
+        else:
+            c_profile = self.nexus_profiles.get(profile, None)
+            if c_profile is None:
+                raise MissingProfileError(
+                    "connection profile not found: '%s'" % profile)
+
+            self.dial_config.update(c_profile)
+
+        # Validate configuration
+        (is_valid, errors) = self.validate_config(self.dial_config)
+        if not is_valid:
+            raise ValidationFailError(
+                "Validation failed: %s" % errors)
+
+        return self.dial_config
+
+    def init_dialer(self):
+        try:
+            self.logger.debug("connection string: %s \n" % self.dial_config)
+            self.nexus_client = NexusClient(**self.dial_config)
+        except Exception as error:
+            self.logger.error("Couldn't instantiate nexus client %s" % error)
```

Comment:
This swallows the error - is that what you want?

I would fail loudly - make sure you reraise the error.